src/playback-watcher.js

```diff
@@ -211,6 +211,24 @@ export default class PlaybackWatcher {
       this.timer_ = setTimeout(this.skipTheGap_.bind(this),
                                difference * 1000,
                                currentTime);
+      return;
+    }
+
+    let currentRange = Ranges.findRange(buffered, currentTime);
+
+    // Sometimes the player can stall for unknown reasons within a contiguous buffered
+    // region with no indication that anything is amiss (seen in Firefox). Seeking to
+    // currentTime is usually enough to kickstart the player.
+    if (currentRange.length) {
```

Comment:
Although shorter segments aren't all too common, the original one could break because it wasn't checking to see if the current time was within a buffered range. This one does check that the current time is within a buffered range. The TIME_FUDGE_FACTOR may be enough of a safety margin from the edge of a buffered region, but it could be worth being a bit safer with how far it is from the edge of its containing region.